server/services/prisonService.test.ts

```diff
@@ -0,0 +1,211 @@
+
+      prisonRegisterStore.getActivePrisons.mockResolvedValue([])
+      prisonRegisterClient.getAllPrisons.mockResolvedValue(allPrisons)
+
+      const moorlandPrisonResponse = aValidPrisonResponse({
+        prisonId: 'MDI',
+        prisonName: 'Moorland (HMP & YOI)',
+        active: true,
+      })
+
+      const expectedPrison = aValidPrison({
+        prisonId: 'MDI',
+      })
+      mockedPrisonMapper.mockReturnValue(expectedPrison)
+
+      // When
+      const actual = await prisonService.getPrisonByPrisonId(prisonId, systemToken)
+
+      // Then
+      expect(actual).toEqual(expectedPrison)
+      expect(prisonRegisterStore.getActivePrisons).toHaveBeenCalled()
+      expect(mockedPrisonMapper).toHaveBeenCalledWith(moorlandPrisonResponse)
+      expect(prisonRegisterClient.getAllPrisons).toHaveBeenCalled()
+      expect(prisonRegisterStore.setActivePrisons).toHaveBeenCalledWith(allPrisons, 1)
+    })
+
+    it('should not get prison by ID given prison does not exist in cache or API', async () => {
+      // Given
+      const prisonId = 'some-unknown-prison-id'
+      const systemToken = 'a-system-token'
+
+      prisonRegisterStore.getActivePrisons.mockResolvedValue(activePrisons)
+      prisonRegisterClient.getAllPrisons.mockResolvedValue(allPrisons)
+
+      // When
+      const actual = await prisonService.getPrisonByPrisonId(prisonId, systemToken)
+
+      // Then
+      expect(actual).toEqual(undefined)
+      expect(prisonRegisterStore.getActivePrisons).toHaveBeenCalled()
+      expect(mockedPrisonMapper).not.toHaveBeenCalled()
+      expect(prisonRegisterClient.getAllPrisons).toHaveBeenCalled()
+      expect(prisonRegisterStore.setActivePrisons).toHaveBeenCalledWith(allPrisons, 1)
+    })
+
+    it('should get prison by ID given retrieving from cache throws an error', async () => {
+      // Given
+      const prisonId = 'MDI'
+      const systemToken = 'a-system-token'
+
+      prisonRegisterStore.getActivePrisons.mockRejectedValue('some-error')
+      prisonRegisterClient.getAllPrisons.mockResolvedValue(allPrisons)
+
+      const moorlandPrisonResponse = aValidPrisonResponse({
+        prisonId: 'MDI',
+        prisonName: 'Moorland (HMP & YOI)',
+        active: true,
+      })
+
+      const expectedPrison = aValidPrison({
+        prisonId: 'MDI',
+      })
+      mockedPrisonMapper.mockReturnValue(expectedPrison)
+
+      // When
+      const actual = await prisonService.getPrisonByPrisonId(prisonId, systemToken)
+
+      // Then
+      expect(actual).toEqual(expectedPrison)
+      expect(prisonRegisterStore.getActivePrisons).toHaveBeenCalled()
+      expect(mockedPrisonMapper).toHaveBeenCalledWith(moorlandPrisonResponse)
+      expect(prisonRegisterClient.getAllPrisons).toHaveBeenCalled()
+      expect(prisonRegisterStore.setActivePrisons).toHaveBeenCalledWith(allPrisons, 1)
+    })
+
+    it('should not get prison by ID given retrieving from cache and API both throw errors', async () => {
```

Comment:
Crikey, I hope this never happens!

Reply:
💯 , agreed; but at least we have it covered 😁 👍